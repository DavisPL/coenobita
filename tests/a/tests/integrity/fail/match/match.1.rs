#![allow(unused_variables)]
#![allow(unused_assignments)]

enum Foo {
    #[cnbt::tag({bin}{bin} struct ({bin}{bin,c}))]
    Bar(i32),
    
    #[cnbt::tag({bin}{bin} struct ({a}{a}, {b}{b}))]
    Baz(i32, i32)
}

fn main() {
    #[cnbt::tag({bin}{bin})]
    let f = Foo::Bar(5);

    match f {
        Foo::Bar(x) => {
            #[cnbt::tag({bin}{bin,c})]
            let x_ = x;
        }

        Foo::Baz(y, z) => {
            #[cnbt::tag({a}{a})]
            let y_ = y;

            #[cnbt::tag({b}{b})]
            let z_ = z;
        }
    }
}