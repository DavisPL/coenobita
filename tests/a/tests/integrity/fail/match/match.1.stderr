error: expected {bin}{bin}, found {b}{b}
  --> tests/integrity/fail/match/match.1.rs:25:22
   |
25 |             let z_ = z;
   |                      ^
